packages/adminrouter/extra/src/test-harness/tests/test_service.py

```diff
@@ -842,3 +843,51 @@ def test_if_ar_sets_correct_useragent_while_resolving_via_mesosdns(
 
         assert len(r_reqs) == 1
         verify_header(r_reqs[0]['headers'], 'User-Agent', 'Master Admin Router')
+
+    @pytest.mark.parametrize(
```

Comment:
Not for now but I'd like to point out that this is an *ideal* use case for [Hypothesis](https://hypothesis.readthedocs.io). That is, a test which says "Give me any string which is not `0, false, no` and make the following assumption" could be written very simply.

❓ Ideally for me There would be a docstring something like:

"""
If `DCOS_SERVICE_REWRITE_REQUEST_URLS` is set to `0`, `false` or `no`, Admin Router will not strip the context path and the upstream request will be made with the same the URL path as the client request has been made. Else...
"""

Even better would be two tests, one with `0, false, no` parameters, and one with a bunch of other strings.

Is it the case that `DCOS_SERVICE_REWRITE_REQUEST_URLS` could be not set?

Reply:
> I get a feeling that introducing Hypothesis here could be a bit of an overkill, plain @pytest.mark.parametrize should just enough. Would you agree?

Strongly "no" - but maybe "too much effort to be worth it right now" which is why I said "Not for now" and wrote the comment just as a brain seed, not as a "do this now" instruction.

Reply:
Discussed via DM, we will pair with Adam this week and see how this could be improved with Hypothesis.

Reply:
We ended up with not implementing it due to https://github.com/pytest-dev/pytest/issues/916. It prevents using pytest fixtures without some extra hacking :(

In overall, I find Hypothesis to be a nice tool. Once the issue is fixed I would like to try to implement test case like these using it.